lib/chef/resource/sysctl.rb

```diff
@@ -116,6 +116,30 @@ def set_sysctl_param(key, value)
           shell_out!("sysctl #{'-e ' if new_resource.ignore_error}-w \"#{key}=#{value}\"")
         end
       end
+
+      private
+
+      # shellout to sysctl to get the current value
+      # ignore missing keys by using '-e'
+      # convert tabs to spaces since sysctl tab deliminates multivalue parameters
+      # strip the newline off the end of the output as well
+      # and compare the value with chef-created config to maintain idempotency
+      #
+      def get_sysctl_value(key)
+        val = shell_out!("sysctl -n -e #{key}").stdout.tr("\t", " ").strip
+        raise unless val == get_sysctld_value(key)
```

Comment:
Isn't this going to give us no current value if the values just don't match?

Reply:
I guess we kind of have to do that since the system is in a 1/2 configured state. We should probably comment above the raise why we have to treat the systctl and sysctld values not matching as a no current value situation.

Reply:
Yes, I referred within list of exceptions and found out this to be the best match according to the [comment](https://github.com/chef/chef/blob/master/lib/chef/exceptions.rb#L127).
Will update the comment above this raise

Reply:
It might be easier to set `value` to `nil` instead of dealing with `current_resource` being `nil`.

i'm not certain i'm wild about the whole `current_value_does_not_exist!` API...

Reply:
@lamont-granquist :+1:  Initially I thought the same, but since value is a required property, it raises a exception at runtime:
```
Chef::Exceptions::ValidationFailed: value is a required property
```

@tas50 : Ive updated the comments as per the suggestions. Could you please take a look and let me know in case of any issues

Reply:
yeah, i see, and based on our bad handling of required properties (#5992, #4596) there's no really good solution here, so it looks like this is the way to fix this today...